Nsec bug and UI UX

itailiors opened the pull request.

itailiors commented:
Branch including:
tests for src/Angor.Test/DerivationOperationsTest.cs
Better stats for each project (and corrected the stats existing)
Fix for Nsec bug that it didn't prompt the password.

and a bit of UI UX (like adding a button to paste address and clear)
![image](https://github.com/user-attachments/assets/c808b9a8-0c6d-4dbc-ae2d-04063d8f1bd2)

dangershony commented on src/Angor/Client/Pages/Investor.razor:
```diff
@@ -45,7 +45,7 @@
                 }
             </button>
-            <button class="btn btn-secondary" @onclick="NavigateToPenalties">View Penalties</button>
+            <button class="btn btn-secondary" @onclick="NavigateToPenalties" data-toggle="tooltip" title="Penalties return control to investors but put Bitcoin on hold for a penalty duration.">View Penalties</button>
```
good we need more of this

dangershony commented:
Can you please fix the conflicts?

dangershony commented on src/Angor/Client/Pages/View.razor:
```diff
@@ -162,43 +163,60 @@
-                        Stages
-                    </h5>
+       
+            <div class="card-body mt-4">
```
this are UI changes that might effect what @miladsoft just did can you check this Milad?

dangershony commented on src/Angor/Client/Pages/View.razor:
```diff
@@ -19,7 +19,8 @@
 @inherits BaseComponent
 <NotificationComponent @ref="notificationComponent" />
-
+<NotificationComponent @ref="notificationComponent" />
```
is there duplicate notification component here?

miladsoft commented:
It seems good.
Some parts of the UI are messed up, but I will fix it.

itailiors commented:
> Enhanced Password Handling: Implemented improved password management for displaying the NSec key. Users are now shown a loading indicator while the NSec key is being generated and displayed. If an incorrect password is entered and the password dialog is closed, no loading indicator is shown. The dialog reappears if the password needs to be re-entered.
> 
> Error Handling: Added comprehensive error handling for incorrect password entries, including displaying an appropriate error message and retrying the password entry process.
> 
> Open in Browse Functionality: Added a new feature to open the project in Browse with the project's public key embedded in the URL. This allows users to quickly navigate to the project details in a new browser tab.
> 
> UI Improvements: Updated the UI to show loading indicators and error messages appropriately based on the password validation and NSec key generation process.

Great changes, loved them.
Thanks!

dangershony commented on src/Angor/Client/Pages/View.razor:
```diff
@@ -454,27 +516,97 @@
+            }
+        }
+        catch (InvalidOperationException)
+        {
```
should we not handle the password check on the popup when entering the password?
I think we do or used to do it.

miladsoft replied:
Yes, it is checked, but if the user does not press the submit button and exits the password entry window by pressing X, then we can no longer check the wrongly entered password.

dangershony commented on src/Angor/Client/Pages/View.razor:
```diff
@@ -454,27 +516,97 @@
+        if (!string.IsNullOrEmpty(pubkey))
+        {
+            // Construct the URL with the pubkey
+            var url = $"https://browse.angor.io/projects/{pubkey}";
```
ah nice.... this will be cool when browse is up and running

dangershony commented on src/Angor/Client/Pages/Wallet.razor:
```diff
@@ -1214,6 +1228,25 @@ else
+        }
+        catch (Exception ex)
+        {
+            Console.Error.WriteLine($"Failed to read clipboard contents: {ex}");
```
use the logging class (do it in next pr)
